src/btf.cpp

```diff
@@ -585,48 +607,33 @@ std::unique_ptr<std::istream> BTF::get_funcs(std::regex *re,
     err = btf_dump__emit_type_decl(dump, t->type, &decl_opts);
     if (err)
     {
-      LOG(ERROR) << "failed to dump type for: " << func_name;
+      LOG(ERROR) << "failed to dump return type for: " << func_name;
       break;
     }
 
-    funcs += "    " + type + " retval;\n";
-#endif
+    params[func_name].push_back(type + " retval");
   }
 
   if (id != (max + 1))
     LOG(ERROR) << "BTF data inconsistency " << id << "," << max;
 
   btf_dump__free(dump);
 
-  return std::make_unique<std::istringstream>(funcs);
-}
-
-void BTF::display_kfunc(std::regex *re, const bool retprobe = false) const
-{
-  if (!has_data())
-    return;
-
-  auto funcs = get_funcs(re, bt_verbose, retprobe ? "kretfunc:" : "kfunc:");
-  if (!funcs)
-    return;
-
-  std::string func_name;
-  while (std::getline(*funcs, func_name))
-  {
-    std::cout << func_name << std::endl;
-  }
+  return params;
+#else
+  LOG(ERROR) << "Could not get kfunc arguments "
+                "(HAVE_LIBBPF_BTF_DUMP_EMIT_TYPE_DECL is not set)" return {};
+  return {};
+#endif
 }
 
-void BTF::display_structs(std::regex *re) const
+std::set<std::string> BTF::get_structs() const
```

Comment:
nit: whichever is ok, but the function is `get_all_structs`?

Reply:
correct, fixed